src/twisted/internet/defer.py

```diff
@@ -53,6 +53,13 @@ class TimeoutError(Exception):
 
 
 
+class NoCurrentResult(BaseException):
```

Comment:
I'd like to see some maintainer-oriented documentation about the decision to make this a `BaseException` instead of an `Exception`.